### PR TITLE
gnupg: add documentation about smartcard setups

### PR DESCRIPTION
**Pull request by jcgruenhage**
The resulting page here is going to be linked in the `gnupg2-scdaemon.INSTALL.msg` which will resolve https://github.com/void-linux/void-packages/issues/38034, together with https://github.com/void-linux/void-packages/pull/38037.

### CODE REVIEW

File: src/config/openpgp/index.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# OpenPGP Tooling
+
+## GnuPG
+
+Void ships both GnuPG legacy (as `gnupg1`) and GnuPG stable (as `gnupg`). A
+package for the LTS branch is currently not available. Details for configuring
```
Comment (ahesford) [inline src/config/openpgp/index.md:6]: Drop the mention of the LTS package; it doesn't add a lot of useful information.

**Review by ahesford — CHANGES_REQUESTED** [review]
I don't see a good reason to separate the "OpenPGP Tooling" from the smart-cart instructions; it would be better to combine these into one document.

File: src/config/openpgp/gnupg-smartcards.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# GnuPG and Smartcards
+
+For using Smartcards such as Yubikeys with GnuPG, there are two backends for
+communicating with them through GnuPG: The internal CCID driver of GnuPG's
+scdaemon, or the PC/SC driver.
+
+## scdaemon with internal CCID driver
+
+By default, scdaemon, which is required for using smartcards with GnuPG, uses
+it's internal CCID driver. For this to work, your smartcard needs to be one of
```
Comment (classabbyamp) [inline src/config/openpgp/gnupg-smartcards.md:10]: `its` at the beginning of the line

Comment (jcgruenhage) [reply src/config/openpgp/gnupg-smartcards.md]: done

**Review by ahesford — APPROVED** [review]

File: src/config/gnupg.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+For using smartcards such as Yubikeys with GnuPG, there are two backends for
+communicating with them through GnuPG: The internal CCID driver of GnuPG's
+scdaemon, or the PC/SC driver.
+
+## scdaemon with internal CCID driver
+
+By default, scdaemon, which is required for using smartcards with GnuPG, uses
+its internal CCID driver. For this to work, your smartcard needs to be one of
+the smartcards in the udev rules
+[here](https://salsa.debian.org/debian/gnupg2/-/blob/debian/unstable/debian/scdaemon.udev)
```
Comment (classabbyamp) [inline src/config/gnupg.md:16]: let's link our version of these rules (https://github.com/void-linux/void-packages/blob/master/srcpkgs/gnupg/files/60-scdaemon.rules)

Comment (jcgruenhage) [reply src/config/gnupg.md:16]: Right, that was moved.

Comment (classabbyamp) [reply src/config/gnupg.md]: ignore it :)

File: src/config/gnupg.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+scdaemon, or the PC/SC driver.
+
+## scdaemon with internal CCID driver
+
+By default, scdaemon, which is required for using smartcards with GnuPG, uses
+its internal CCID driver. For this to work, your smartcard needs to be one of
+the smartcards in the udev rules
+[here](https://github.com/void-linux/void-packages/blob/master/srcpkgs/gnupg/files/60-scdaemon.rules)
+and you need to be using logind or be a member of the plugdev group, if you're
+not using logind. If these two condition are fulfilled and you don't have pcscd
```
[the remark this answers is not in the document]
Comment (jcgruenhage) [reply src/config/gnupg.md]: applied, reformatted and pushed.

**Comment by lemmi** [comment]
Thanks everyone for handling this so promptly!

**Pull request merged**